### ISSUE TITLE
The map of LinkedHashMapSecuredJsCache  cache too many entry

### ISSUE THREAD

**Issue by Bearin** [javadelight/delight-nashorn-sandbox#98]
I have one instance of NashornSandbox in my application, and config like this:
`        sandbox = NashornSandboxes.create();
        sandbox.setMaxCPUTime(100);
        sandbox.setMaxMemory(100 * 1024 * 1024L);
        sandbox.allowNoBraces(false);
        sandbox.setMaxPreparedStatements(30);
        sandbox.setExecutor(Executors.newFixedThreadPool(4));
`
call sandbox.eval in multi thread, then I dump jvm heap file,  
The JsSanitizer maintain more than 700,000 linkedHashMap$Entry, more than 180 MB memory been used. Thr LRU lose effectiveness?

**Comment by mxro** [comment]
Hi @Bearin , thanks for reporting this issue!

Hmm that's interesting - are these related to the `SecuredJsCache`? In that case, could you try with `sandbox.setMaxPreparedStatements(30);`. Any other details you could share about the dump?

**Comment by Bearin** [comment]
> Hi @Bearin , thanks for reporting this issue!
> 
> Hmm that's interesting - are these related to the `SecuredJsCache`? In that case, could you try with `sandbox.setMaxPreparedStatements(30);`. Any other details you could share about the dump?

Yes, I's related to the LinkedHashMapSecuredJsCache in NashornSandboxImpl.JsSanitizer; LinkedHashMap is not thread safe,  maybe you can make it recurrence in a muitl-thread test.
